src/sas/qtgui/Perspectives/Fitting/FittingWidget.py

```diff
@@ -2049,10 +2049,13 @@ def fromModelToQModel(self, model_name):
         self.shell_names = self.shellNamesList()
 
         # Update the QModel
-        new_rows = FittingUtilities.addParametersToModel(self.model_parameters, self.kernel_module, self.is2D)
+        FittingUtilities.addParametersToModel(
```

Comment:
I pass the model and view objects in because I should be initializing AssociatedComboBox objects while processing parameters in FittingUtilities, and callling view.setIndexWidget **after** the items are added to the model. The only way I can cleanly instantiate the combo boxes in FittingUtilities before applying them to the view during row appending in FittingWidget is by using QStandardItem's data storage, but the snippet below demonstrates a PyQt bug preventing me from doing that:

```python
import sys
from PyQt5 import QtGui, QtWidgets

def f():
    # externally set item's data to a combo box
    item = QtGui.QStandardItem()
    addComboBox(item)

    # attempt to manipulate the combo box - the object has been destructed (partially?)
    # and AttributeError is raised
    cbox = item.data()
    print(cbox.currentText())

def addComboBox(item):
    cbox = QtWidgets.QComboBox()
    cbox.addItems(["option1", "option2"])
    item.setData(cbox)

    # manipulate
    print(cbox.currentText())

if __name__ == "__main__":
    app = QtWidgets.QApplication(sys.argv)
    f()
```

The other option is to get it to return a dict(?) of combo boxes and their related items, but that's even worse than simply passing an extra couple of optional parameters to the FittingUtilities methods.